### PR TITLE
Add support for lowercase HTTP methods in route() method

### PR DESCRIPTION
**Pull request by ashikkabeer**
#### Checklist

- [ x] run `npm run test` and `npm run benchmark`
- [x ] tests and/or benchmarks are included
- [ x] documentation is changed or added
- [x ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by jsumners — CHANGES_REQUESTED** [review]
Please add unit tests to cover this.

**Comment by ashikkabeer** [comment]
@jsumners I don't really know how to write unit tests. Can you help me with some information on that? I saw there are many frameworks for that. What framwork should i use ?

**Comment by Uzlopak** [comment]
See /test/route.js

You basically have to do:

```
test('route - lowercase method', t => {
    t.plan(3)
      fastify.route({
        method: 'get',
        url: '/lowercase',
        handler: function (req, reply) {
          reply.send({ hello: 'world' })
        }
      })
      sget({
        method: 'GET',
        url: 'http://localhost:' + fastify.server.address().port + '/lowercase'
      }, (err, response, body) => {
        t.error(err)
        t.equal(response.statusCode, 200)
        t.same(JSON.parse(body), { hello: 'world' })
      })
  })
```

I didnt test it, wrote it on my mobile. But you should get the gist...

File: fastify.js [hunk 1]
```diff
@@ -280,6 +280,10 @@ function fastify (options) {
     route: function _route (options) {
       // we need the fastify object that we are producing so we apply a lazy loading of the function,
       // otherwise we should bind it after the declaration
+      
+      if (typeof options.method === 'string') {
+            options.method = options.method.toUpperCase();
```
Comment (Eomm) [inline fastify.js:285]: Note that this `method` could be an array, so we should support `method: ['get', 'POST']` too

**Comment by ashikkabeer** [comment]
@Uzlopak I got this error  FAIL ​ test/route.test.js 1 failed of 187 7s
 ✖ Cannot read property 'port' of null

**Comment by Uzlopak** [comment]
@ashikkabeer 

Sry, i saw now, that you commented. I thought, that you closed out of frustration. 

I opened #4750 as a follow up PR. It was imho a little bit more complicated to implement it properly.

**Comment by ashikkabeer** [comment]
I wasn't frustrated or anything. I closed it accidentally. I'm starting to do OpenSource contributions and this is my first code contribution attempt. :)

**Pull request closed without merging**